src/main/java/world/jnc/invsync/util/serializer/module/mod/CyclicSyncModule.java

```diff
@@ -0,0 +1,143 @@
+    }
+
+    @Nonnull
+    @Override
+    public ItemStack getStackInSlot(int slot) {
+      return inventory.getStackInSlot(slot);
+    }
+
+    @Nonnull
+    @Override
+    public ItemStack insertItem(int slot, @Nonnull ItemStack stack, boolean simulate) {
+      return ItemStack.EMPTY;
+    }
+
+    @Nonnull
+    @Override
+    public ItemStack extractItem(int slot, int amount, boolean simulate) {
+      return ItemStack.EMPTY;
+    }
+
+    @Override
+    public int getSlotLimit(int slot) {
+      return 0;
+    }
+  }
+
+  @UtilityClass
+  private static class Helper {
+    private static final DataQuery CRAFTING_CAPABILITY = DataQuery.of("crafting_capability");
+    private static final DataQuery INVENTORY_CAPABILITY = DataQuery.of("inventory_capability");
+    private static final DataQuery INVENTORY = DataQuery.of("inventory");
+
+    private static DataView serialize(Player player, DataView container) {
+      EntityPlayer nativePlayer = NativeInventorySerializer.getNativePlayer(player);
+      DataContainer serializedData = DataContainer.createNew(DataView.SafetyMode.ALL_DATA_CLONED);
+
+      IPlayerExtendedProperties props = CapabilityRegistry.getPlayerProperties(nativePlayer);
+      serializedData.set(CRAFTING_CAPABILITY, props.hasInventoryCrafting());
+      serializedData.set(INVENTORY_CAPABILITY, props.hasInventoryExtended());
+
+      InventoryPlayerExtended inventory =
+          UtilPlayerInventoryFilestorage.getPlayerInventory(
+              NativeInventorySerializer.getNativePlayer(player));
+      serializedData.set(
+          INVENTORY, NativeInventorySerializer.serializeInventory(new InventoryAdapter(inventory)));
+
+      container.set(THIS, serializedData);
+      return container;
+    }
+
+    private static void deserialize(Player player, DataView container) {
+      Optional<DataView> serializedData = container.getView(THIS);
+      if (getDebug()) {
+        getLogger().info("\t\tisPresent:");
+        getLogger().info("\t\t\tserializedData:\t" + serializedData.isPresent());
+      }
+
+      if (serializedData.isPresent()) {
+        EntityPlayer nativePlayer = NativeInventorySerializer.getNativePlayer(player);
+        IPlayerExtendedProperties props = CapabilityRegistry.getPlayerProperties(nativePlayer);
+
+        props.setInventoryCrafting(serializedData.get().getBoolean(CRAFTING_CAPABILITY).get());
+        props.setInventoryExtended(serializedData.get().getBoolean(INVENTORY_CAPABILITY).get());
+        CapabilityRegistry.syncServerDataToClient((EntityPlayerMP) nativePlayer);
+
+        // Create a new inventory to deserialize into. This way if anything goes wrong, the current
+        // extended inventory isn't overwritten.
+        InventoryPlayerExtended inventory = new InventoryPlayerExtended(nativePlayer);
+
+        boolean fail =
+            NativeInventorySerializer.deserializeInventory(
+                serializedData.get().getViewList(INVENTORY).get(), new InventoryAdapter(inventory));
```

Comment:
Store the queries in an Optional please and print them during debug. Debug message shlould also be pretty late.